src/sheet_metal_conus.py

```diff
@@ -57,29 +84,130 @@ def __init__(self):
-        return longColor
-
-    #A*256^0 + B*256^1 + G*256^2 + R*256^3
-    def getColorString(self, longColor):
-        " convert the long into a #RRGGBB color value "
-        longColor = self.unsignedLong(longColor)
+    def getColorString(self, longColor, verbose=False):
+        """ Convert the long into a #RRGGBB color value
+            - verbose=true pops up value for us in defaults
+            conversion back is A + B*256^1 + G*256^2 + R*256^3
+        """
+        if verbose: inkex.debug("%s ="%(longColor))
+        longColor = long(longColor)
+        if longColor <0: longColor = long(longColor) & 0xFFFFFFFF
         hexColor = hex(longColor)[2:-3]
-        hexColor = hexColor.rjust(6, '0')
-        return '#' + hexColor.upper()
+        hexColor = '#' + hexColor.rjust(6, '0').upper()
+        if verbose: inkex.debug("  %s for color default value"%(hexColor))
+        return hexColor
+
+    # Marker arrows
+    def makeMarkerstyle(self, name, rotate):
+        " Markers added to defs for reuse "
+        defs = self.xpathSingle('/svg:svg//svg:defs')
+        if defs == None:
+            defs = inkex.etree.SubElement(self.document.getroot(),inkex.addNS('defs','svg'))
+        marker = inkex.etree.SubElement(defs ,inkex.addNS('marker','svg'))
+        marker.set('id', name)
+        marker.set('orient', 'auto')
+        marker.set('refX', '0.0')
+        marker.set('refY', '0.0')
+        marker.set('style', 'overflow:visible')
+        marker.set(inkex.addNS('stockid','inkscape'), name)
+
+        arrow = inkex.etree.Element("path")
+        # definition of arrows in beautiful DIN-shapes:
+        if name.startswith('ArrowDIN-'):
+            if rotate:
+                arrow.set('d', 'M 8,0 -8,2.11 -8,-2.11 z')
+            else:
+                arrow.set('d', 'M -8,0 8,-2.11 8,2.11 z')
+        if name.startswith('ArrowDINout-'):
+            if rotate:
+                arrow.set('d', 'M 0,0 16,2.11 16,0.5 26,0.5 26,-0.5 16,-0.5 16,-2.11 z')
+            else:
+                arrow.set('d', 'M 0,0 -16,2.11 -16,0.5 -26,0.5 -26,-0.5 -16,-0.5 -16,-2.11 z')
+        arrow.set('style', 'fill:#000000;stroke:none')
+        marker.append(arrow)
+
+    def set_arrow_dir(self, option, style):
+        if option=='inside':
+            # inside
+            self.arrowlen = 6.0
+            style['marker-start'] = 'url(#ArrowDIN-start)'
+            style['marker-end'] = 'url(#ArrowDIN-end)'
+            self.makeMarkerstyle('ArrowDIN-start', False)
+            self.makeMarkerstyle('ArrowDIN-end', True)
+        else:
+            # outside
+            self.arrowlen = 0
+            style['marker-start'] = 'url(#ArrowDINout-start)'
+            style['marker-end'] = 'url(#ArrowDINout-end)'
+            self.makeMarkerstyle('ArrowDINout-start', False)
+            self.makeMarkerstyle('ArrowDINout-end', True)
 
+    def drawDimArc(self, center, start, end, radius, style, parent, gap=0, lowside=True):
```

Comment:
This just draws the arc for angle dimension with arrows at either end. If gap defined it will break into two arcs with suitable arrow markers at one end